wa/utils/android.py

```diff
@@ -132,3 +135,68 @@ def _pod_upgrade_v1(pod):
         pod['_pod_version'] = pod.get('_pod_version', 1)
         return pod
 
+
+class ApkInfoCache:
+
+    @staticmethod
+    def _check_env():
+        if not os.path.exists(settings.cache_directory):
+            os.makedirs(settings.cache_directory)
+
+    def __init__(self, path=settings.apk_info_cache_file):
+        self._check_env()
+        self.path = path
+        self.last_modified = None
+        self.cache = {}
+        self._initialised = os.path.exists(path)
+        if self._initialised:
+            self._update_cache()
+
+    def store(self, apk_info, apk_id, overwrite=True):
+        self._update_cache()
+        if apk_id in self.cache and not overwrite:
+            raise ValueError('ApkInfo for {} is already in cache.'.format(apk_info.path))
+        self.cache[apk_id] = apk_info.to_pod()
+        with lock_file(self.path):
+            write_pod(self.cache, self.path)
+            self.last_modified = os.stat(self.path)
+
+    def get_info(self, key):
+        self._update_cache()
+        pod = self.cache.get(key)
+
+        info = ApkInfo.from_pod(pod) if pod else None
+        return info
+
+    def _update_cache(self):
+        if not self._initialised:
+            return
+        if self.last_modified != os.stat(self.path):
```

Comment:
Can the `os.stat` not result in a `FileNotFoundError` being raised when an `ApkInfoCache` is created in a new environment for the first time, and `store` has not previously been called, so that the cache file does not exist?

Reply:
That was meant to have been prevented by the use of `_initialised` however it was not set correctly so have removed and added a check to see if the file exists explicitly.